crates/experimental-api/src/rpc.rs

```diff
@@ -291,6 +299,28 @@ impl gen::Rpc for Context {
         key: StorageKey,
         block_id: BlockId,
     ) -> std::result::Result<Felt, jsonrpc::Error> {
+        let l1_block_num = self.node.read().await.l1_block_number as i64;
```

Comment:
This block seems like a good candidate for a method/function extraction (and maybe relevant unit-testing if it makes sense).

Reply:
Migrated to `impl Context`. Added proper unit tests